README.md

```diff
@@ -1,27 +1,67 @@
 
 Random sources are committed to the [`RandomBeaconHistory` contract](./contracts/RandomBeaconHistory.cdc) by the service
-account at the end of every block. These random sources are catalogued chronologically, extending historically for every
-associated block height to the initial commitment height.
+account at the end of every block. The RandomBeaconHistory contract provides a convenient archive, where for each past
+block height (starting Nov 2023) the respective 'source of randomness' can be retrieved.
 
-Used on their own, these random sources are not safe. In other words, using the random source in your contract without
-the framing of a commit-reveal mechanism would enable non-trusted callers to condition their interactions with your contract on the
+When used naively, `revertibleRandom` as well as the [`RandomBeaconHistory` contract](./contracts/RandomBeaconHistory.cdc)
+are subject to post-selection attacks from transactions.
+In simple terms, using the random source in your contract without
+the protection of a commit-reveal mechanism would enable non-trusted callers to condition their interactions with your contract on the
 random result. In the context of a random coin toss, I could revert my transaction if I didn't win - not a very fair
 game.
 
 To achieve non-revertible randomness, the contract should be structured to resolve in two phases:
 
 1. Commit - Caller commits to the resolution of their bet with some yet unknown source of randomness (i.e. in the
    future)
-2. Reveal - Caller can then reveal the result of their bet
+2. Reveal - Caller can then resolve the result of their bet once the source of randomness is available in the `RandomBeaconHistory` with a separate transaction.
+   From a technical perspective, this could also be called "resolving transaction", because the transaction simply executes the smart contract with the locked-in
+   inputs, whose output all parties committed to accept in the previous phase.
 
-Though a caller could still condition the revealing transaction on the coin flip result, they've already incurred the
-cost of their bet and would gain nothing by doing so.
+Though a caller could still condition the revealing transaction on the coin flip result, all the inputs influencing the bet's outcome
+have already been fixed (the source of randomness being the last one that is only generated after the commit transaction concluded.)
+Conceptually, this corresponds to owning a winning (or loosing) lottery ticket, where the numbers have already been published,
+but the ticket has not been handed in to the lottery company to affirm the win (or loss).
+All that the resolving transaction (reveal phase) is doing is affirming the win or loss.
+The ticket owner could revert their resolving transaction. Though that does not change whether the ticket won or lost. Furthermore, the player has already
+incurred the cost of their bet and gains nothing by reverting the reveal step.
 
 ## References
```

Comment:
A good opportunity to link the new doc as a reference, closing out #7 - https://developers.flow.com/build/advanced-concepts/randomness

Reply:
I revised and extended the `References` section and renamed it. It is now called [`Further Reading`](https://github.com/onflow/random-coin-toss/blob/1a948c66c0f59c11212851c1b7ff2ce5c31b2663/README.md?plain=1#L66-L75). In response to your comment, I added the following reference as the first item in the list
> We recommend the **Flow developer documentation** [**_Advanced Concepts → Flow VRF_**](https://developers.flow.com/build/advanced-concepts/randomness) for important concepts and context on safely using Flow's VRF.  

Please resolve this comment of you feel your suggestion is addressed.